src/prefect/utilities/schema_tools/validation.py

```diff
@@ -253,5 +253,24 @@ def preprocess_schema(
                 process_properties(
                     definition["properties"], required_fields, allow_none_with_default
                 )
+            # Allow block types to be referenced by their id
+            if "block_type_slug" in definition:
+                schema["definitions"][definition["title"]] = {
+                    "oneOf": [
+                        definition,
+                        {
+                            "type": "object",
+                            "properties": {
+                                "$ref": {
+                                    "type": "string",
+                                    "format": "uuid",
+                                },
+                            },
+                            "required": [
+                                "$ref",
+                            ],
+                        },
+                    ]
+                }
```

Comment:
The reference format used when nested block documents in other block documents are `{"$ref": {"block_document_id": block_document_id } }.` I'd prefer to use that same format here for consistency and to leave room for other reference types in the future (like using the block slug as @cicdw suggested).

Reply:
Do i understand correctly that you request changes to the UI itself? i tried to fit current solution to current UI functions, since changes to UI would require more work. I can do that, but it would take more time.

Reply:
Added additional schema to enable your proposed reference formats. I kept the current format, as it is used by Deployment Flow Run Create form. I hope i understood your request correctly.

Reply:
Thanks for adding that! I forgot that we'd need to update the UI to populate the backend data to match that format. I can create a separate issue to allow different reference shapes.